workers/unity/Packages/io.improbable.worker.sdk/IOStream.cs

```diff
@@ -4,45 +4,67 @@
 
 namespace Improbable.Worker.CInterop
 {
+    [Flags]
     public enum OpenMode
     {
-        /* Opens the stream in the default mode. */
-        OpenModeDefault = 0x00,
+        /**
+         * Allow read operations on the stream. Read operations always occur at the read position, which
+         * is initialized to the beginning of the stream.
+         */
+        OpenModeRead = 1,
+
+        /**
+         * Allow write operations on the stream. Write operations always occur at the write position,
+         * which is initialized to the end of the stream.
+         */
+        OpenModeWrite = 2,
+
+        /**
+         * Truncates any existing content upon opening. If not set, writes are appended to the end of the
+         * stream's existing content.
+         */
+        OpenModeTruncate = 4,
+
+        /**
+         * Specify that writes should be appended to the stream's existing content, if any exists.
+         */
+        OpenModeAppend = 8,
     }
 
     public sealed unsafe class IOStream : IDisposable
     {
-        private readonly CIO.StreamHandle stream;
+        internal readonly CIO.StreamHandle Stream;
 
         private IOStream(CIO.StreamHandle stream)
         {
-            this.stream = stream;
+            Stream = stream;
         }
 
         /// <inheritdoc cref="IDisposable"/>
         public void Dispose()
         {
-            stream.Dispose();
+            Stream.Dispose();
         }
 
         public static IOStream CreateRingBufferStream(uint capacity)
         {
             return new IOStream(CIO.CreateRingBufferStream(capacity));
         }
 
-        public static IOStream CreateFileStream(string fileName, OpenMode openMode)
+        public static IOStream CreateFileStream(string fileName,
+            OpenMode? openMode = OpenMode.OpenModeRead | OpenMode.OpenModeWrite | OpenMode.OpenModeTruncate)
```

Comment:
Why is `openMode` a nullable with a default?

Reply:
Oops, I can remove the `?`